channeld/channel.c

```diff
@@ -2206,8 +2200,8 @@ int main(int argc, char *argv[])
 
 	init_peer_crypto_state(peer, &peer->pcs);
 
-	/* Read init_channel message sync. */
-	init_channel(peer);
+	daemon_conn_init(peer, &peer->master, REQ_FD, req_in, master_gone);
+	status_setup_async(&peer->master);
 
```

Comment:
So, this whole patch is the opposite of what I had to do to fix the wait-for-master-reply logic; remove masterfd from ccan/io and always use it sync (also, a nice simplification).

I put an assert() in at the top of init_channel(): assert(!(fcntl(REQ_FD, F_GETFL) & O_NONBLOCK)); and it's not triggering; I can't see how we're getting a nonblocking fd here, and so this fix seems voodoo to me....

Reply:
I tried several times to set the fd to blocking either before passing it to the subdaemon or directly in `main` of the subdaemon, both times it did not change the empty reads from the fd. I'm also at a loss about why this kept happening.

Reply:
Ok, rebasing and removing the async commit, since it is not really related.